command/agent/dns_test.go

```diff
@@ -2125,8 +2140,96 @@ func TestDNS_ServiceLookup_MaxResponses(t *testing.T) {
 			t.Fatalf("err: %v", err)
 		}
 
-		if len(in.Answer) != 3 {
-			t.Fatalf("should receive 3 answers for AAAA")
+		if len(in.Answer) != configUDPAnswerLimit {
+			t.Fatalf("should receive %d answers for AAAA, received: %d", configUDPAnswerLimit, len(in.Answer))
+		}
+	}
+}
+
+func TestDNS_ServiceLookup_UDPAnswerLimit(t *testing.T) {
```

Comment:
This one is kind of similar to the others. Did you intend to have the user limit exceed the hard limit and see that get enforced here?